File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamapps/callback/RedeemGuestPassResponseCallback.java
package in.dragonbra.javasteam.steam.handlers.steamapps.callback;

import in.dragonbra.javasteam.enums.EResult;
import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2.CMsgClientRedeemGuestPassResponse;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;
import in.dragonbra.javasteam.types.JobID;

public class RedeemGuestPassResponseCallback extends CallbackMsg {

    private EResult result;

    private Integer packageID;

    private Integer mustOwnAppID;

    public RedeemGuestPassResponseCallback(JobID jobID, CMsgClientRedeemGuestPassResponse.Builder msg) {
        setJobID(jobID);

        this.result = EResult.from(msg.getEresult());
        this.packageID = msg.getPackageId();
        this.mustOwnAppID = msg.getMustOwnAppid();
    }

    /**
     * @return Result of the operation
     */
    public EResult getResult() {
        return result;
    }

    /**
     * @return Result of the operation
     */
    public Integer getPackageID() {
        return packageID;
    }

    /**
     * @return App ID which must be owned to activate this guest pass.
     */
    public Integer getMustOwnAppID() {
        return mustOwnAppID;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamapps/callback/RedeemGuestPassResponseCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steamapps.callback

import `in`.dragonbra.javasteam.base.ClientMsgProtobuf
import `in`.dragonbra.javasteam.base.IPacketMsg
import `in`.dragonbra.javasteam.enums.EResult
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2.CMsgClientRedeemGuestPassResponse
import `in`.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg

/**
 * This callback is received in response to activating a guest pass or a gift.
 */
@Suppress("MemberVisibilityCanBePrivate")
class RedeemGuestPassResponseCallback(packetMsg: IPacketMsg) : CallbackMsg() {

    /**
     * Gets Result of the operation
     */
    val result: EResult

    /**
     * Gets Result of the operation
     */
    val packageID: Int

    /**
     * Gets App ID which must be owned to activate this guest pass.
     */
    val mustOwnAppID: Int

    init {
        val redeemedGuestPass = ClientMsgProtobuf<CMsgClientRedeemGuestPassResponse.Builder>(
            CMsgClientRedeemGuestPassResponse::class.java,
            packetMsg
        )
        val msg = redeemedGuestPass.body

        jobID = redeemedGuestPass.targetJobID
        result = EResult.from(msg.eresult)
        packageID = msg.packageId
        mustOwnAppID = msg.mustOwnAppid
    }
}
